core/new-gui/src/app/dashboard/component/feature-container/user-dictionary-section/ngbd-modal-resource-add/ngbd-modal-resource-add.component.ts

```diff
@@ -60,4 +67,55 @@ export class NgbdModalResourceAddComponent {
     }
     this.onClose();
   }
+  public uploadFile(){
+    //For "upload" button. Upload the file in the queue and then clear the queue
+    //FileItem: https://github.com/valor-software/ng2-file-upload/blob/development/src/file-upload/file-item.class.ts
+    //typeof queue -> [FileItem]
+    //typeof FileItem._file -> File
+    this.uploader.queue.forEach((item)=>this.userDictionaryService.uploadDictionary(item._file));
+    this.uploader.clearQueue()
+  }
+  public removeFile(item:any){
+    //For "delete" button. Remove the specific file
+    //typeof item -> FileItem
+    //when the user remove one file which is not text type, drop the number of counter with one
+    if (!item._file.type.includes("text")){
+      this.invalidFileNumbe--;
+    }
+    item.remove();
+    //after one file has been removed, check if duplication still exist
+    this.checkDuplicateFiles()
+  }
+  public haveFileOver(e:any):void {
+    // typeof e -> bool
+    //when user drag file over the area, this function will be called with a bool
+    this.haveDropZoneOver = e;
+  }
+  public checkThisFileDuplicate(filename:string){
+    //passed by single file name. check if this file is duplicated in the array
+    return this.duplicateFile.includes(filename);
+  }
+  public checkDuplicateFiles(){
+    //go through the uploader to check if there exist file with the same name, store the duplicate file in this.duplicateFile
+    let filesArray = this.uploader.queue.map(item => item._file.name);
+    let duplicate:any=[];
+    for (var i=0; i<filesArray.length;i++){
+      if (filesArray.indexOf(filesArray[i]) !== i && !duplicate.includes(filesArray[i])){
+        duplicate.push(filesArray[i]);
+        }
+    }
```

Comment:
declare your variable as const if not modified later. When finding the duplicates, use `filter` function provided in JavaScript.